projects/portal/src/app/pages/faucet/faucet.usecase.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { TestBed } from '@angular/core/testing';
+import { FaucetUseCaseService } from './faucet.usecase.service';
+import { combineLatest, of } from 'rxjs';
+import { ConfigService } from '../../models/config.service';
+
+const setup = (props?: {
+  mockConfigService?: any;
+}) => {
+  // Mock Values
+  const mockConfig = {
+    extension: {
+      faucet: [
+        {
+          denom: 'test_denom1',
+          network: 'test_network',
+          faucetURL: 'www/test.url',
+          hasFaucet: true,
+          creditAmount: 1200,
+          maxCredit: 1500,
+        },
+        {
+          denom: 'test_denom2',
+          network: 'test_network',
+          faucetURL: 'www/test.url',
+          hasFaucet: true,
+          creditAmount: 800,
+          maxCredit: 1000,
+        },
+      ],
+    },
+  }
+
+  // Mock Service
+  const mockConfigService = {
+    config$: of(mockConfig),
+  };
+
+  // Setup TestBed
+  TestBed.configureTestingModule({
+    providers: [
+      FaucetUseCaseService,
+      { provide: ConfigService, useValue: { ...mockConfigService, ...props?.mockConfigService } },
+    ],
+  });
+  const service = TestBed.inject(FaucetUseCaseService);
+
+  return {
+    service,
+    //mockConfigService,
+    mockConfig,
+  };
+};
+
+
+
+describe('FaucetUseCaseService when ConfigService returns a valid value', () => {
+  it('should be created', () => {
+    const { service } = setup();
+    expect(service).toBeTruthy();
+  });
+
+  test('denoms$ getter return a valid value', (done) => {
+    const { service } = setup();
+    service.denoms$.subscribe((values) => {
+      expect(values?.every((v) => v.includes("test_denom"))).toBeTruthy();
+      done();
+    });
+  });
+
+  it('faucetURL$ method return a valid value', (done) => {
+    const { service, mockConfig } = setup();
+    service.faucetURL$(of(mockConfig.extension.faucet[0].denom))
+      .subscribe((value) => {
+        expect(value).toBe(mockConfig.extension.faucet[0].faucetURL);
+        done();
```

Comment:
I wonder it is enough to test only 1st element.

Reply:
I think this is enough for what we want to test in this case.